File: Lib/FPVR/Entities/Attitude.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FPVR
{
    /// <summary>
    /// Represents an orientation.
    /// </summary>
    public struct Attitude
    {
        #region Constructors
        /// <summary>
        /// Initializes a new <see cref="Attitude"/>.
        /// </summary>
        /// <param name="yaw">
        /// The yaw in degrees.
        /// </param>
        /// <param name="pitch">
        /// The pitch in degrees.
        /// </param>
        /// <param name="roll">
        /// The roll in degrees.
        /// </param>
        public Attitude(double yaw, double pitch, double roll)
        {
            this.Yaw = yaw;
            this.Pitch = pitch;
            this.Roll = roll;
        }
        #endregion // Constructors

        #region Public Members
        /// <summary>
        /// The yaw in degrees.
        /// </summary>
        public double Yaw;

        /// <summary>
        /// The pitch in degrees.
        /// </summary>
        public double Pitch;

        /// <summary>
        /// The roll in degrees.
        /// </summary>
        public double Roll;
        #endregion // Public Members
    }
}

File: Android/FPVR/app/src/main/java/com/solersoft/fpvr/fpvrlib/Attitude.java
package com.solersoft.fpvr.fpvrlib;

/**
 * Created by jbienz on 5/20/2015.
 */
public class Attitude
{
    //region Constructors

    /**
     * Initializes a new Attitude instance where all angles are zero.
     */
    public Attitude(){}

    /**
     * Initializes a new Attitude instance.
     * @param pitch The pitch in degrees.
     * @param roll The roll in degrees.
     * @param yaw The yaw in degrees.
     */
    public Attitude(double pitch, double roll, double yaw)
    {
        this.pitch = pitch;
        this.roll = roll;
        this.yaw = yaw;
    }
    //endregion

    //region Public Members
    /**
     * The yaw in degrees.
     */
    public double yaw=0;

    /**
     * The pitch in degrees.
     */
    public double pitch=0;

    /**
     * The roll in degrees.
     */
    public double roll=0;
    //endregion

}
